app/src/androidTest/java/com/github/sdpcoachme/SignupActivityTest.kt

```diff
@@ -106,16 +107,18 @@ open class SignupActivityTest {
         inputUserInfo(exceptionUser)
 
         // Wait for activity to send to database
-        scenario.onActivity { activity ->
-            var exceptionThrown = false
-            activity.databaseStateSending.handle { _, _ ->
-                exceptionThrown = true
-                // Recover from exception
-                null
-            }.get(10, SECONDS)
-            // Make sure exception was thrown
-            TestCase.assertTrue(exceptionThrown)
+        lateinit var databaseStateSending: CompletableFuture<Void>
+        scenario.onActivity {
+            databaseStateSending = it.databaseStateSending
         }
+        var exceptionThrown = false
+        databaseStateSending.handle { _, _ ->
+            exceptionThrown = true
+            // Recover from exception
+            null
+        }.get(10, SECONDS)
```

Comment:
Yes, for all those get calls, it is important to have them outside to make sure there are no deadlocks.